File: frontend/src/microservices/hotel/objects/SearchSuggestionPlannerResponse.ts
export type SearchSuggestionPlannerResponse = {
  resourceType: string
  value: string
  title: string
  subtitle: string
}

export const searchSuggestionPlannerResponseFromJson = (json: string): SearchSuggestionPlannerResponse =>
  JSON.parse(json) as SearchSuggestionPlannerResponse

export const searchSuggestionPlannerResponseToJson = (value: SearchSuggestionPlannerResponse): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/hotel/objects/SearchSuggestionPlannerResponse.scala
package com.typesafe.travel.hotel.objects

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class SearchSuggestionPlannerResponse(
    resourceType: String,
    value: String,
    title: String,
    subtitle: String
)

object SearchSuggestionPlannerResponse:
  given sourceEncoder: Encoder[SearchSuggestionPlannerResponse] = deriveEncoder
  given sourceDecoder: Decoder[SearchSuggestionPlannerResponse] = deriveDecoder
